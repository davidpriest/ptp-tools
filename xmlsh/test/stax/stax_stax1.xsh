import commands stax=stax


r=stax:newEventReader(../../samples/data/books.xml)


while [ stax:hasNext($r) ] 
do
	
	e=stax:nextEvent($r)
	type=stax:getEventType($e)
	echo event type $type

	case	$type in
		START_ELEMENT ) 	echo "  " stax:getName($e) ;
			if [ stax:getName($e) = "ITEM" ] ; then
				a=stax:getAttribute($e  "CAT")
				xtype $a
				echo CAT is a.getValue()
			fi
			;;
		CHARACTERS    ) echo " " stax:getData($e) ;;
	esac
	
done

stax:closeReader $r 


