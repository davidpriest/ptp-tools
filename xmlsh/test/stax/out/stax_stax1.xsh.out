event type START_DOCUMENT
event type PROCESSING_INSTRUCTION
event type DTD
event type START_ELEMENT
   BOOKLIST
event type CHARACTERS
  

event type START_ELEMENT
   BOOKS
event type CHARACTERS
  
	
event type START_ELEMENT
   ITEM
event type CHARACTERS
  
    	
event type START_ELEMENT
   TITLE
event type CHARACTERS
  Pride and Prejudice
event type END_ELEMENT
event type CHARACTERS
  
    	
event type START_ELEMENT
   AUTHOR
event type CHARACTERS
  Jane Austen
event type END_ELEMENT
event type CHARACTERS
  
    	
event type START_ELEMENT
   PUBLISHER
event type CHARACTERS
  Modern Library
event type END_ELEMENT
event type CHARACTERS
  
    	
event type START_ELEMENT
   PUB-DATE
event type CHARACTERS
  2002-12-31
event type END_ELEMENT
event type CHARACTERS
  
    	
event type START_ELEMENT
   LANGUAGE
event type CHARACTERS
  English
event type END_ELEMENT
event type CHARACTERS
  
    	
event type START_ELEMENT
   PRICE
event type CHARACTERS
  4.95
event type END_ELEMENT
event type CHARACTERS
  
    	
event type START_ELEMENT
   QUANTITY
event type CHARACTERS
  187
event type END_ELEMENT
event type CHARACTERS
  
    	
event type START_ELEMENT
   ISBN
event type CHARACTERS
  0679601686
event type END_ELEMENT
event type CHARACTERS
  
    	
event type START_ELEMENT
   PAGES
event type CHARACTERS
  352
event type END_ELEMENT
event type CHARACTERS
  
    	
event type START_ELEMENT
   DIMENSIONS
event type CHARACTERS
  8.3 5.7 1.1
event type END_ELEMENT
event type CHARACTERS
  
    	
event type START_ELEMENT
   WEIGHT
event type CHARACTERS
  6.1
event type END_ELEMENT
event type CHARACTERS
  
	
event type END_ELEMENT
event type CHARACTERS
  
	
event type START_ELEMENT
   ITEM
event type CHARACTERS
  
		
event type START_ELEMENT
   TITLE
event type CHARACTERS
  Wuthering Heights
event type END_ELEMENT
event type CHARACTERS
  
		
event type START_ELEMENT
   AUTHOR
event type CHARACTERS
  Charlotte BrontÃ«
event type END_ELEMENT
event type CHARACTERS
  
		
event type START_ELEMENT
   PUBLISHER
event type CHARACTERS
  Penguin Classics
event type END_ELEMENT
event type CHARACTERS
  
		
event type START_ELEMENT
   PUB-DATE
event type CHARACTERS
  2002-12-31
event type END_ELEMENT
event type CHARACTERS
  
		
event type START_ELEMENT
   LANGUAGE
event type CHARACTERS
  English
event type END_ELEMENT
event type CHARACTERS
  
		
event type START_ELEMENT
   PRICE
event type CHARACTERS
  6.58
event type END_ELEMENT
event type CHARACTERS
  
		
event type START_ELEMENT
   QUANTITY
event type CHARACTERS
  113
event type END_ELEMENT
event type CHARACTERS
  		
		
event type START_ELEMENT
   ISBN
event type CHARACTERS
  0141439556
event type END_ELEMENT
event type CHARACTERS
  
		
event type START_ELEMENT
   PAGES
event type CHARACTERS
  430
event type END_ELEMENT
event type CHARACTERS
  
    
event type START_ELEMENT
   DIMENSIONS
event type CHARACTERS
  1.0 5.2 7.8
event type END_ELEMENT
event type CHARACTERS
  
		
event type START_ELEMENT
   WEIGHT
event type CHARACTERS
  11.2
event type END_ELEMENT
event type CHARACTERS
  
	
event type END_ELEMENT
event type CHARACTERS
  
	
event type START_ELEMENT
   ITEM
event type CHARACTERS
  
		
event type START_ELEMENT
   TITLE
event type CHARACTERS
  Tess of the d'Urbervilles
event type END_ELEMENT
event type CHARACTERS
  
		
event type START_ELEMENT
   AUTHOR
event type CHARACTERS
  Thomas Hardy
event type END_ELEMENT
event type CHARACTERS
  
		
event type START_ELEMENT
   PUBLISHER
event type CHARACTERS
  Bantam Classics
event type END_ELEMENT
event type CHARACTERS
  
		
event type START_ELEMENT
   PUB-DATE
event type CHARACTERS
  1984-05-01
event type END_ELEMENT
event type CHARACTERS
  
		
event type START_ELEMENT
   LANGUAGE
event type CHARACTERS
  English
event type END_ELEMENT
event type CHARACTERS
  
		
event type START_ELEMENT
   PRICE
event type CHARACTERS
  4.95
event type END_ELEMENT
event type CHARACTERS
  
		
event type START_ELEMENT
   QUANTITY
event type CHARACTERS
  85
event type END_ELEMENT
event type CHARACTERS
  		
		
event type START_ELEMENT
   ISBN
event type CHARACTERS
  0553211684
event type END_ELEMENT
event type CHARACTERS
  
		
event type START_ELEMENT
   PAGES
event type CHARACTERS
  480
event type END_ELEMENT
event type CHARACTERS
  
    
event type START_ELEMENT
   DIMENSIONS
event type CHARACTERS
  6.8 4.2 0.8
event type END_ELEMENT
event type CHARACTERS
  
		
event type START_ELEMENT
   WEIGHT
event type CHARACTERS
  7.7
event type END_ELEMENT
event type CHARACTERS
  
	
event type END_ELEMENT
event type CHARACTERS
  
	
event type START_ELEMENT
   ITEM
event type CHARACTERS
  
		
event type START_ELEMENT
   TITLE
event type CHARACTERS
  Jude the Obscure
event type END_ELEMENT
event type CHARACTERS
  
		
event type START_ELEMENT
   AUTHOR
event type CHARACTERS
  Thomas Hardy
event type END_ELEMENT
event type CHARACTERS
  
		
event type START_ELEMENT
   PUBLISHER
event type CHARACTERS
  Penguin Classics
event type END_ELEMENT
event type CHARACTERS
  
		
event type START_ELEMENT
   PUB-DATE
event type CHARACTERS
  1998-09-01
event type END_ELEMENT
event type CHARACTERS
  
		
event type START_ELEMENT
   LANGUAGE
event type CHARACTERS
  English
event type END_ELEMENT
event type CHARACTERS
  
		
event type START_ELEMENT
   PRICE
event type CHARACTERS
  4.95
event type END_ELEMENT
event type CHARACTERS
  
		
event type START_ELEMENT
   QUANTITY
event type CHARACTERS
  129
event type END_ELEMENT
event type CHARACTERS
  		
		
event type START_ELEMENT
   ISBN
event type CHARACTERS
  0140435387
event type END_ELEMENT
event type CHARACTERS
  
		
event type START_ELEMENT
   PAGES
event type CHARACTERS
  528
event type END_ELEMENT
event type CHARACTERS
  
    
event type START_ELEMENT
   DIMENSIONS
event type CHARACTERS
  7.8 5.2 0.9
event type END_ELEMENT
event type CHARACTERS
  
		
event type START_ELEMENT
   WEIGHT
event type CHARACTERS
  10.9
event type END_ELEMENT
event type CHARACTERS
  
	
event type END_ELEMENT
event type CHARACTERS
  
	
event type START_ELEMENT
   ITEM
event type CHARACTERS
  
		
event type START_ELEMENT
   TITLE
event type CHARACTERS
  The Big Over Easy
event type END_ELEMENT
event type CHARACTERS
  
		
event type START_ELEMENT
   AUTHOR
event type CHARACTERS
  Jasper Fforde
event type END_ELEMENT
event type CHARACTERS
  
		
event type START_ELEMENT
   PUBLISHER
event type CHARACTERS
  Hodder & Stoughton
event type END_ELEMENT
event type CHARACTERS
  
		
event type START_ELEMENT
   PUB-DATE
event type CHARACTERS
  2005-07-11
event type END_ELEMENT
event type CHARACTERS
  
		
event type START_ELEMENT
   LANGUAGE
event type CHARACTERS
  English
event type END_ELEMENT
event type CHARACTERS
  
		
event type START_ELEMENT
   PRICE
event type CHARACTERS
  16.47
event type END_ELEMENT
event type CHARACTERS
  
		
event type START_ELEMENT
   QUANTITY
event type CHARACTERS
  129
event type END_ELEMENT
event type CHARACTERS
  		
		
event type START_ELEMENT
   ISBN
event type CHARACTERS
  0340835672
event type END_ELEMENT
event type CHARACTERS
  
		
event type START_ELEMENT
   PAGES
event type CHARACTERS
  346
event type END_ELEMENT
event type CHARACTERS
  
    
event type START_ELEMENT
   DIMENSIONS
event type CHARACTERS
  22.5 18.0 3.5
event type END_ELEMENT
event type CHARACTERS
  
		
event type START_ELEMENT
   WEIGHT
event type CHARACTERS
  390
event type END_ELEMENT
event type CHARACTERS
  
	
event type END_ELEMENT
event type CHARACTERS
  
	
event type START_ELEMENT
   ITEM
event type CHARACTERS
  
		
event type START_ELEMENT
   TITLE
event type CHARACTERS
  The Eyre Affair
event type END_ELEMENT
event type CHARACTERS
  
		
event type START_ELEMENT
   AUTHOR
event type CHARACTERS
  Jasper Fforde
event type END_ELEMENT
event type CHARACTERS
  
		
event type START_ELEMENT
   PUBLISHER
event type CHARACTERS
  Penguin
event type END_ELEMENT
event type CHARACTERS
  
		
event type START_ELEMENT
   PUB-DATE
event type CHARACTERS
  2003-02-25
event type END_ELEMENT
event type CHARACTERS
  
		
event type START_ELEMENT
   LANGUAGE
event type CHARACTERS
  English
event type END_ELEMENT
event type CHARACTERS
  
		
event type START_ELEMENT
   PRICE
event type CHARACTERS
  16.47
event type END_ELEMENT
event type CHARACTERS
  
		
event type START_ELEMENT
   QUANTITY
event type CHARACTERS
  129
event type END_ELEMENT
event type CHARACTERS
  		
		
event type START_ELEMENT
   ISBN
event type CHARACTERS
  0142001805
event type END_ELEMENT
event type CHARACTERS
  
		
event type START_ELEMENT
   PAGES
event type CHARACTERS
  384
event type END_ELEMENT
event type CHARACTERS
  
    
event type START_ELEMENT
   DIMENSIONS
event type CHARACTERS
  7.8 5.0 0.9
event type END_ELEMENT
event type CHARACTERS
  
		
event type START_ELEMENT
   WEIGHT
event type CHARACTERS
  9.0
event type END_ELEMENT
event type CHARACTERS
  
	
event type END_ELEMENT
event type CHARACTERS
  


event type END_ELEMENT
event type CHARACTERS
  

event type START_ELEMENT
   CATEGORIES
event type CHARACTERS
  
    
event type START_ELEMENT
   CATEGORY
event type END_ELEMENT
event type CHARACTERS
  
    
event type START_ELEMENT
   CATEGORY
event type END_ELEMENT
event type CHARACTERS
  
    
event type START_ELEMENT
   CATEGORY
event type END_ELEMENT
event type CHARACTERS
  

event type END_ELEMENT
event type CHARACTERS
  

event type END_ELEMENT
event type END_DOCUMENT
